webapp/exercise_admin/forms.py

```diff
@@ -115,25 +112,22 @@ def _add_file_fields(self, file_id, default_lang, lang_list):
             else:
                 self.fields[default_name_field] = forms.CharField(max_length=255, required=False, strip=True)
 
-            self.fields[description_field] = forms.CharField(required=False, strip=True)
-
-    #TODO: Move to main form!
-    def _add_link_fields(self, file_id, default_lang, lang_list):
-        return
+            self.fields[description_field] = forms.CharField(required=False, strip=True)        
 
-    def _clean_duplicates_of_field(self, field_name, field_val, model_field, cleaned_data, lang_list):
-        field_prefix = "instance_file_{}".format(model_field)
+    def _clean_duplicates_of_field(self, field_name, field_val, field_prefix, model_field, cleaned_data, lang_list):
         model_field_readable = model_field.replace("_", " ")
-        
+
         for lang_code, _ in lang_list:
-            if field_name.startswith(field_prefix) and field_name.endswith(lang_code) and field_val:
-                for k, v in cleaned_data.copy().items():
-                    if k.startswith(field_prefix) and k.endswith(lang_code) and v == field_val and k != field_name:
-                        error_msg = "Duplicate {field} in language {lang}!".format(field=model_field_readable, lang=lang_code)
-                        error_code = "duplicate_{}".format(model_field)
-                        field_error = forms.ValidationError(error_msg, code=error_code)
-                        self.add_error(field_name, field_error)
-        
+            if lang_code in field_name:
```

Comment:
This is going to cause problems. Consider lang_codes such as 'fi' with field names with the string 'file' for example. (And there are hundreds or so different language codes.)